nni/compression/experiment/experiment.py

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import logging
+from pathlib import Path
+from typing import Any, Callable, List, Optional, Union
+
+import torch
+from torch.nn import Module
+from torch.optim import Optimizer
+
+from nni.compression.experiment.config.utils import cc_cv2ss
+from nni.experiment import Experiment, ExperimentConfig
+from .config import CompressionConfig, CompressionVessel
+
+_logger = logging.getLogger('nni.experiment')
+
+
+class CompressionExperiment(Experiment):
+    def __init__(self, config_or_platform: Union[ExperimentConfig, str, List[str]],
+                 compression_config: CompressionConfig, model: Module, finetuner: Callable[[Module], None],
+                 evaluator: Callable[[Module], float], dummy_input: Optional[Any],
+                 trainer: Optional[Callable[[Module, Optimizer, Callable[[Any, Any], Any]], None]],
+                 optimizer: Optional[Optimizer], criterion: Optional[Callable[[Any, Any], Any]],
+                 device: str):
+        super().__init__(config_or_platform)
+
+        self.compression_config = compression_config
+        assert all([model, finetuner, evaluator])
+        assert all([trainer, optimizer, criterion]) or not any([trainer, optimizer, criterion])
+        torch.save(model.state_dict(), Path(self.config.trial_code_directory, 'nni_model_state_dict.pth'))
```

Comment:
I don't suggest putting "garbages" arbitrarily, as this is usually not git-ignored. How about saving it into nni-experiments folder?